src/code/z_camera.c

```diff
@@ -1,373 +1,7988 @@
+    VecSph sp6C;
+    VecSph atToEye;
+    VecSph atToEyeNext;
+    PosRot* playerPosRot = &camera->trackActorPosRot;
+    Special5FixedData* fixedData = CAM_GET_STATIC_DATA(Special5);
+    Special5DynamicData* dynamicData = CAM_GET_DYNAMIC_DATA(Special5);
+    f32 rand;
+    f32 trackHeight = Camera_GetTrackedActorHeight(camera);
+
+    if (RELOAD_PARAMS) {
+        CameraModeValue* values = sCameraSettings[camera->setting].cameraModes[camera->mode].values;
+        f32 yNormal = (0.8f - ((68.0f / trackHeight) * -0.2f));
+
+        fixedData->yOffset = (READ_SCALED_STATIC_DATA_VAL * trackHeight) * yNormal;
+        fixedData->eyeDist = READ_STATIC_DATA_VAL;
+        fixedData->minDistForRot = READ_STATIC_DATA_VAL;
+        fixedData->pitch = DEGF_TO_BINANG(READ_STATIC_DATA_VAL);
+        fixedData->fovTarget = READ_STATIC_DATA_VAL;
+        fixedData->atMaxLERPScale = READ_SCALED_STATIC_DATA_VAL;
+        fixedData->timerInit = READ_STATIC_DATA_VAL;
+        fixedData->flags = READ_STATIC_DATA_VAL;
+    }
+
+    OLib_Vec3fDiffToVecSphGeo(&atToEye, at, eye);
+    OLib_Vec3fDiffToVecSphGeo(&atToEyeNext, at, eyeNext);
+    Actor_GetWorld(&spA8, camera->target);
+
+    sCameraInterfaceFlags = fixedData->flags;
+
+    if (camera->actionFuncState == 0) {
+        camera->actionFuncState++;
+        dynamicData->animTimer = fixedData->timerInit;
+    }
+
+    if (dynamicData->animTimer > 0) {
+        dynamicData->animTimer--;
+    } else if (dynamicData->animTimer == 0) {
+        if ((camera->target == NULL) || (camera->target->update == NULL)) {
+            camera->target = NULL;
+            return true;
+        }
+
+        dynamicData->animTimer--;
+        if (fixedData->minDistForRot < OLib_Vec3fDist(&spA8.pos, &playerPosRot->pos)) {
+            sp6C.yaw = playerPosRot->rot.y;
+            sp6C.pitch = -playerPosRot->rot.x;
+            sp6C.r = 20.0f;
+            OLib_VecSphAddToVec3f(&sp7C.pos, &spA8.pos, &sp6C);
+            func_800CBC84(camera, at, &sp7C, 0);
+            OLib_Vec3fToVecSphGeo(&sp6C, &sp7C.norm);
+            spA4 = SUB16(playerPosRot->rot.y, sp6C.yaw);
+            sp74.r = fixedData->eyeDist;
+            rand = Rand_ZeroOne();
+            sp74.yaw = BINANG_ROT180(playerPosRot->rot.y) +
+                       (s16)(spA4 < 0 ? -(s16)(0x1553 + (s16)(rand * 2730.0f)) : (s16)(0x1553 + (s16)(rand * 2730.0f)));
+            sp74.pitch = fixedData->pitch;
+            OLib_VecSphAddToVec3f(eyeNext, &spA8.pos, &sp74);
+            *eye = *eyeNext;
+            Camera_BGCheck(camera, &spA8.pos, eye);
+        }
+    }
+
+    Camera_CalcAtDefault(camera, &atToEyeNext, fixedData->yOffset, 0);
+    camera->fov = Camera_LERPCeilF(fixedData->fovTarget, camera->fov, camera->atLERPStepScale * 0.05f, 0.1f);
+    camera->roll = Camera_LERPCeilS(0, camera->roll, 0.5f, 5);
+    camera->atLERPStepScale = Camera_ClampLERPScale(camera, fixedData->atMaxLERPScale);
+    return true;
+}
+
+s32 Camera_Special6(Camera* camera) {
+    return 
```

Comment:
Wouldn't make more sense to treat this function as a boolean?
```suggestion
s32 Camera_IsDbgCamEnabled(void) {
    return false;
}
```